src/ntfsck.c

```diff
@@ -1162,6 +1162,49 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
 	goto next_inode;
 }
 
+static void ntfsck_check_mft_record_unused(ntfs_volume *vol, s64 mft_num)
+{
+	MFT_RECORD *mrec;
+	u16 seq_no;
+	s64 pos = mft_num * vol->mft_record_size;
+	s64 count = 512;
+
+	mrec = ntfs_malloc(count);
```

Comment:
where mrec is freed?

Reply:
Need to free it. Thanks for pointing out.

Reply:
I'm confused, I think you use ntfs_mst_pread() to read mft record not ntfs_attr_pread().

Reply:
Why? we touch only flags and seq_no in mft entry.